src/systemActs/PayloadTypes.ts

```diff
@@ -11,10 +11,17 @@
  * permissions and limitations under the License.
  */
 
+export interface RenderedPayload<T> {
```

Comment:
not bad, but I'm not sure if it adds enough to have an interface.  e.g. because ValueChanged has various rendered things. i.e. it doesn't seem a really strong abstraction.

consider dropping the interface in interests of simplicity / readability.

Reply:
Updated, I created an interface since we might have other params related to rendering in future, but yeah for now I can remove and add it on each payloads.